pandas/core/array_algos/masked_reductions.py

```diff
@@ -100,3 +100,12 @@ def min(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
 
 def max(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
     return _minmax(np.max, values=values, mask=mask, skipna=skipna)
+
+
+def mean(values: np.ndarray, mask: np.ndarray, skipna: bool = True):
+    if not values.size:
+        return libmissing.NA
+    _sum = _sumprod(np.sum, values=values, mask=mask, skipna=skipna)
+    count = np.count_nonzero(~mask)
```

Comment:
We also need to worry about division by zero (a case of non-empty but all missing should also return NA)

Reply:
I could add mask.all() then?

Reply:
yes exactly